python/lsst/pipe/tasks/makeCoaddTempExp.py

```diff
@@ -58,9 +58,150 @@ class MakeCoaddTempExpConfig(CoaddBaseTask.ConfigClass):
+
+    @section pipe_tasks_makeCoaddTempExp_Purpose  Description
+
+    Warp and optionally PSF-Match calexps onto a common projection, by
+    performing the following operations:
+    - Group calexps by visit/run
+    - For each visit, generate a Warp by calling method @ref makeTempExp.
+      makeTempExp loops over the visit's calexps calling @ref WarpAndPsfMatch
+      on each visit
+
+    The result is a `directWarp` (and/or optionally a `psfMatchedWarp`).
+
+    @section pipe_tasks_makeCoaddTempExp_Initialize  Task Initialization
+
+    @copydoc \_\_init\_\_
+
+    This task has no special keyword arguments.
+
+    @section pipe_tasks_makeCoaddTempExp_IO  Invoking the Task
+
+    This task is primarily designed to be run from the command line.
+
+    The main method is `run`, which takes a single butler data reference for the patch(es)
+    to process.
+
+    @copydoc run
+
+    @copydoc WarpType
+
+    @section pipe_tasks_makeCoaddTempExp_Config  Configuration parameters
+
+    See @ref MakeCoaddTempExpConfig and parameters inherited from
+    @link lsst.pipe.tasks.coaddBase.CoaddBaseConfig CoaddBaseConfig @endlink
+
+    @subsection pipe_tasks_MakeCoaddTempExp_psfMatching Guide to PSF-Matching Configs
+
+    To make `psfMatchedWarps`, select `config.makePsfMatched=True`. The subtask
+    @link lsst.ip.diffim.modelPsfMatch.ModelPsfMatchTask ModelPsfMatchTask @endlink
+    is responsible for the PSF-Matching, and its config is accessed via `config.warpAndPsfMatch.psfMatch`.
+    The optimal configuration depends on aspects of dataset: the pixel scale, average PSF FWHM and
+    dimensions of the PSF kernel. These configs include the matching kernel size, the requested model PSF,
+    padding of the science PSF thumbnail and spatial sampling frequency of the PSF.
+    The following list presents the most common problems that arise from a misconfigured ModelPsfMatchTask
+    and corresponding solutions. All assume the default Alard-Lupton kernel, with configs accessed via
+    ```config.warpAndPsfMatch.psfMatch.kernel['AL']```.
+
+    *Troublshooting PSF-Matching Configuration:*
+    - Matched PSFs look boxy: The matching kernel is too small. _Increase the matching kernel size.
+        For example:_
+
+            config.warpAndPsfMatch.psfMatch.kernel['AL'].kernelSize=27  # default 21
+
+        Note that increasing the kernel size also increases runtime.
+    - Matched PSFs look ugly (dipoles, quadropoles, donuts): unable to find good solution
+        for matching kernel. _Provide the matcher with more data by either increasing
+        the spatial sampling by decreasing the spatial cell size,_
+
+            config.warpAndPsfMatch.psfMatch.kernel['AL'].sizeCellX = 64  # default 128
+            config.warpAndPsfMatch.psfMatch.kernel['AL'].sizeCellY = 64  # default 128
+
+        _or increasing the padding around the Science PSF, for example:_
+
+            config.warpAndPsfMatch.psfMatch.autoPadPsfTo=1.6  # default 1.4
```

Comment:
This autoPadPsf and autoPadPsfTo configuration parameters are a bit confusing. I think I get them and their necessity, but it might be worth explaining them a bit here.